.github/workflows/sonar_analysis.yml

```diff
@@ -0,0 +1,39 @@
+name: Build
+
+on:
+  # Trigger analysis when pushing in master or pull requests, and when creating
+  # a pull request.
+  push:
+    branches:
+      - main
+  pull_request:
+      types: [opened, synchronize, reopened]
+jobs:
+  build:
+    name: SonarQube Analysis
+    runs-on: ubuntu-latest
+    permissions: read-all
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          fetch-depth: 0  # Shallow clones should be disabled for a better relevancy of analysis
+      - name: Set up JDK 17
+        uses: actions/setup-java@v1
+        with:
+          java-version: 17
+      - name: Cache SonarQube packages
```

Comment:
`SonarQube` or `SonarCloud`?

Reply:
Let's call it SonarQube as that's the main tool. SonarCloud is their SaaS product